dbms/src/Storages/DeltaMerge/Filter/IntegerSet.cpp

```diff
@@ -0,0 +1,609 @@
+        return std::make_shared<ValueSet<UInt16>>(values);
+    case TypeIndex::DateTime:
+        return std::make_shared<ValueSet<UInt32>>(values);
+    case TypeIndex::Enum8:
+        return std::make_shared<ValueSet<Int8>>(values);
+    case TypeIndex::Enum16:
+        return std::make_shared<ValueSet<Int16>>(values);
+    case TypeIndex::MyDate:
+    case TypeIndex::MyDateTime:
+    case TypeIndex::MyTimeStamp:
+        return std::make_shared<ValueSet<UInt64>>(values);
+    case TypeIndex::MyTime:
+        return std::make_shared<ValueSet<Int64>>(values);
+    default:
+        return nullptr;
+    }
+}
+
+IntegerSetPtr IntegerSet::createLessRangeSet(TypeIndex type_index, Field max, bool not_included)
+{
+    switch (type_index)
+    {
+    case TypeIndex::UInt8:
+        return std::make_shared<RangeSet<UInt8>>(0, max.get<UInt8>() - not_included);
+    case TypeIndex::UInt16:
+        return std::make_shared<RangeSet<UInt16>>(0, max.get<UInt16>() - not_included);
+    case TypeIndex::UInt32:
+        return std::make_shared<RangeSet<UInt32>>(0, max.get<UInt32>() - not_included);
+    case TypeIndex::UInt64:
+        return std::make_shared<RangeSet<UInt64>>(0, max.get<UInt64>() - not_included);
+    case TypeIndex::Int8:
+        return std::make_shared<RangeSet<Int8>>(std::numeric_limits<Int8>::min(), max.get<Int8>() - not_included);
+    case TypeIndex::Int16:
+        return std::make_shared<RangeSet<Int16>>(std::numeric_limits<Int16>::min(), max.get<Int16>() - not_included);
+    case TypeIndex::Int32:
+        return std::make_shared<RangeSet<Int32>>(std::numeric_limits<Int32>::min(), max.get<Int32>() - not_included);
+    case TypeIndex::Int64:
+        return std::make_shared<RangeSet<Int64>>(std::numeric_limits<Int64>::min(), max.get<Int64>() - not_included);
+    case TypeIndex::Date:
+        return std::make_shared<RangeSet<UInt16>>(0, max.get<UInt16>() - not_included);
+    case TypeIndex::DateTime:
+        return std::make_shared<RangeSet<UInt32>>(0, max.get<UInt32>() - not_included);
+    case TypeIndex::Enum8:
+        return std::make_shared<RangeSet<Int8>>(std::numeric_limits<Int8>::min(), max.get<Int8>() - not_included);
+    case TypeIndex::Enum16:
+        return std::make_shared<RangeSet<Int16>>(std::numeric_limits<Int16>::min(), max.get<Int16>() - not_included);
+    case TypeIndex::MyDate:
+    case TypeIndex::MyDateTime:
+    case TypeIndex::MyTimeStamp:
+        return std::make_shared<RangeSet<UInt64>>(0, max.get<UInt64>() - not_included);
+    case TypeIndex::MyTime:
+        return std::make_shared<RangeSet<Int64>>(std::numeric_limits<Int64>::min(), max.get<Int64>() - not_included);
+    default:
+        return nullptr;
+    }
+}
+
+IntegerSetPtr IntegerSet::createGreaterRangeSet(TypeIndex type_index, Field min, bool not_included)
+{
+    switch (type_index)
+    {
+    case TypeIndex::UInt8:
+        return std::make_shared<RangeSet<UInt8>>(min.get<UInt8>() + not_included, std::numeric_limits<UInt8>::max());
```

Comment:
ditto, could we run into a case that `min.get<UInt8> == std::numeric_limits<UInt8>::max()`?